Modules/xSharePoint/DSCResources/MSFT_xSPSearchContentSource/MSFT_xSPSearchContentSource.psm1

```diff
@@ -0,0 +1,365 @@
+        [parameter(Mandatory = $false)] [System.Boolean]  $ContinuousCrawl,
+        [parameter(Mandatory = $false)] [Microsoft.Management.Infrastructure.CimInstance] $IncrementalSchedule,
+        [parameter(Mandatory = $false)] [Microsoft.Management.Infrastructure.CimInstance] $FullSchedule,
+        [parameter(Mandatory = $false)] [ValidateSet("Normal","High")] [System.String] $Priority,
+        [parameter(Mandatory = $false)] [System.UInt32]   $LimitPageDepth,
+        [parameter(Mandatory = $false)] [System.UInt32]   $LimitServerHops,
+        [parameter(Mandatory = $true)]  [ValidateSet("Present","Absent")] [System.String] $Ensure,
+        [parameter(Mandatory = $false)] [System.Boolean]  $Force,
+        [parameter(Mandatory = $false)] [System.Management.Automation.PSCredential] $InstallAccount
+    )
+
+    switch ($ContentSourceType) {
+        "SharePoint" {
+            if ($PSBoundParameters.ContainsKey("LimitPageDepth") -eq $true) { throw "Parameter LimitPageDepth is not valid for SharePoint content sources" }
+            if ($PSBoundParameters.ContainsKey("LimitServerHops") -eq $true) { throw "Parameter LimitServerHops is not valid for SharePoint content sources" }
+            if ($ContinuousCrawl -eq $true -and $PSBoundParameters.ContainsKey("IncrementalSchedule") -eq $true) { throw "You can not specify an incremental crawl schedule on a content source that will use continous crawl" }
+            if ($CrawlSetting -eq "Custom") { throw "Parameter 'CrawlSetting' can only be set to custom for website content sources" }
+        }
+        "Website" {
+            if ($PSBoundParameters.ContainsKey("ContinuousCrawl") -eq $true) { throw "Parameter ContinuousCrawl is not valid for SharePoint content sources" }
+            if ($PSBoundParameters.ContainsKey("LimitServerHops") -eq $true) { throw "Parameter LimitServerHops is not valid for SharePoint content sources" }
+        }
+        "FileShare" {
+            if ($PSBoundParameters.ContainsKey("LimitPageDepth") -eq $true) { throw "Parameter LimitPageDepth is not valid for SharePoint content sources" }
+            if ($PSBoundParameters.ContainsKey("LimitServerHops") -eq $true) { throw "Parameter LimitServerHops is not valid for SharePoint content sources" }
+            if ($CrawlSetting -eq "Custom") { throw "Parameter 'CrawlSetting' can only be set to custom for website content sources" }
+        }
+    }
+    
+    $CurrentValues = Get-TargetResource @PSBoundParameters
+    
+    if ($ContentSourceType -ne $CurrentValues.ContentSourceType -and $Force -eq $false) {
+        throw "The content type can not be changed from '$($CurrentValues.ContentSourceType)' to '$ContentSourceType' without deleting and adding it again. Specify 'Force = `$true' in order to allow DSC to do this, or manually remove the existing content source and re-run the configuration."
```

Comment:
I would change the message to "The content source type" instead of "The content type". This might cause confusion with regular content types.